### PR TITLE
[Storage] Fix remove Azure File Directory fail with pipeline

### LINKED ISSUE #11596: Piping a Directory into Remove-AzStorageDirectory doesn't remove directory
## Description

When piping a Directory into Remove-AzStorageDirectory doesn't remove the directory and doesn't fail.

## Steps to reproduce

I have some code like this:

```powershell
    $ArchiveRootDirectory = Get-AzStorageFile -Share $AzStorageShare -Path $ArchivePath
    $ArchiveDirectories = $ArchiveRootDirectory | Get-AzStorageFile

    $ArchiveDirectories |
    ForEach-Object {
        $_ |
        Get-AzStorageFile |
        ForEach-Object {
            W ('## Removing file "'+ $_.Name + '".')
            $_ | Remove-AzStorageFile
        }

        W ('## Removing directory "'+ $_.Name + '".')
        $_ | Remove-AzStorageDirectory
    }
```

## Environment data

Azure DevOps Azure PowerShell deploy task:

```
2020-04-14T10:25:45.5059485Z ==============================================================================
2020-04-14T10:25:45.5059867Z Task         : Azure PowerShell
2020-04-14T10:25:45.5060171Z Description  : Run a PowerShell script within an Azure environment
2020-04-14T10:25:45.5060462Z Version      : 4.159.12
2020-04-14T10:25:45.5060692Z Author       : Microsoft Corporation
2020-04-14T10:25:45.5061008Z Help         : https://aka.ms/azurepowershelltroubleshooting
2020-04-14T10:25:45.5061360Z ==============================================================================
```

## Verbose output

```
2020-04-14T10:23:27.9874579Z 
2020-04-14T10:23:27.9897359Z 
2020-04-14T10:23:28.0001916Z ServiceClient               : Microsoft.Azure.Storage.File.CloudFileClient
2020-04-14T10:23:28.0002559Z Uri                         : https://<StorageAccountName>.file.core.windows.net/projects/Apps/PTO/Archive/6938.28047
2020-04-14T10:23:28.0004066Z StorageUri                  : Primary = 
2020-04-14T10:23:28.0004856Z                               'https://<StorageAccountName>.file.core.windows.net/projects/Apps/PTO/Archive/6938.28047'; 
2020-04-14T10:23:28.0005470Z                               Secondary = ''
2020-04-14T10:23:28.0006491Z SnapshotQualifiedUri        : https://<StorageAccountName>.file.core.windows.net/projects/Apps/PTO/Archive/6938.28047
2020-04-14T10:23:28.0007529Z SnapshotQualifiedStorageUri : Primary = 
2020-04-14T10:23:28.0008579Z                               'https://<StorageAccountName>.file.core.windows.net/projects/Apps/PTO/Archive/6938.28047'; 
2020-04-14T10:23:28.0009049Z                               Secondary = ''
2020-04-14T10:23:28.0010078Z Properties                  : Microsoft.Azure.Storage.File.FileDirectoryProperties
2020-04-14T10:23:28.0010760Z Metadata                    : {}
2020-04-14T10:23:28.0011349Z FilePermission              : 
2020-04-14T10:23:28.0011992Z Share                       : Microsoft.Azure.Storage.File.CloudFileShare
2020-04-14T10:23:28.0012734Z Parent                      : Microsoft.Azure.Storage.File.CloudFileDirectory
2020-04-14T10:23:28.0013386Z Name                        : 6938.28047
2020-04-14T10:23:28.0013914Z 
2020-04-14T10:23:28.0033008Z 
2020-04-14T10:23:28.0056624Z 
2020-04-14T10:23:28.0179148Z VERBOSE: Performing the operation "Remove directory" on target 
2020-04-14T10:23:28.0182935Z VERBOSE: "Apps/PTO/Archive/6938.28047/Microsoft.Azure.Storage.File.CloudFileDirectory".: 6938.28047
```

## Workaround

Changing the invocation to `Remove-AzStorageDirectory -Directory $_, I get this output and the directory is removed:

```
2020-04-14T10:26:13.0559485Z 
2020-04-14T10:26:13.0591799Z 
2020-04-14T10:26:13.0689317Z ServiceClient               : Microsoft.Azure.Storage.File.CloudFileClient
2020-04-14T10:26:13.0690091Z Uri                         : https://<StorageAccountName>.file.core.windows.net/projects/Apps/PTO/Archive/6938.28047
2020-04-14T10:26:13.0690736Z StorageUri                  : Primary = 
2020-04-14T10:26:13.0694679Z                               'https://<StorageAccountName>.file.core.windows.net/projects/Apps/PTO/Archive/6938.28047'; 
2020-04-14T10:26:13.0695213Z                               Secondary = ''
2020-04-14T10:26:13.0695814Z SnapshotQualifiedUri        : https://<StorageAccountName>.file.core.windows.net/projects/Apps/PTO/Archive/6938.28047
2020-04-14T10:26:13.0696286Z SnapshotQualifiedStorageUri : Primary = 
2020-04-14T10:26:13.0696776Z                               'https://<StorageAccountName>.file.core.windows.net/projects/Apps/PTO/Archive/6938.28047'; 
2020-04-14T10:26:13.0697235Z                               Secondary = ''
2020-04-14T10:26:13.0697671Z Properties                  : Microsoft.Azure.Storage.File.FileDirectoryProperties
2020-04-14T10:26:13.0699751Z Metadata                    : {}
2020-04-14T10:26:13.0700099Z FilePermission              : 
2020-04-14T10:26:13.0700514Z Share                       : Microsoft.Azure.Storage.File.CloudFileShare
2020-04-14T10:26:13.0701177Z Parent                      : Microsoft.Azure.Storage.File.CloudFileDirectory
2020-04-14T10:26:13.0701675Z Name                        : 6938.28047
2020-04-14T10:26:13.0701966Z 
2020-04-14T10:26:13.0716960Z 
2020-04-14T10:26:13.0740968Z 
2020-04-14T10:26:13.0842146Z VERBOSE: Performing the operation "Remove directory" on target "Apps/PTO/Archive/6938.28047".
```

### PR DESCRIPTION
**Pull request by blueww**
## Description

To fix #11596:
Parameter Path has type string, and enable ValueFromPipeline.
But since "ValueFromPipeline" is pipeline by type, and string is a very common type, so will get wrong value with enabled "ValueFromPipeline".
We should not enabled "ValueFromPipeline" for Path with common type string.
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Storage/Storage.Management/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,8 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Fix remove Azure File Directory fail with pipeline
```
Comment (wyunchi-ms) [inline src/Storage/Storage.Management/ChangeLog.md:21]: 1. Please add the contents "Fixed https://github.com/Azure/azure-powershell/issues/11596".
1. We think `Fixed` is consistent with other logs.

Comment (blueww) [reply src/Storage/Storage.Management/ChangeLog.md]: change log fixed

**Comment by blueww** [comment]
There's an error reported in static analysis:

> "Microsoft.Azure.PowerShell.Cmdlets.Storage.dll","Microsoft.WindowsAzure.Commands.Storage.File.Cmdlet.RemoveAzureStorageDirectory","Remove-AzStorageDirectory","0","1050","The parameter set 'Directory' for cmdlet 'Remove-AzStorageDirectory' has been removed.","Add parameter set 'Directory' back to cmdlet 'Remove-AzStorageDirectory'."

But the parameter set is not removed. Why the error reported?

**Comment by wyunchi-ms** [comment]
> There's an error reported in static analysis:
> 
> > "Microsoft.Azure.PowerShell.Cmdlets.Storage.dll","Microsoft.WindowsAzure.Commands.Storage.File.Cmdlet.RemoveAzureStorageDirectory","Remove-AzStorageDirectory","0","1050","The parameter set 'Directory' for cmdlet 'Remove-AzStorageDirectory' has been removed.","Add parameter set 'Directory' back to cmdlet 'Remove-AzStorageDirectory'."
> 
> But the parameter set is not removed. Why the error reported?

The error message is not accurate enough. It thinks that the origin parameter set can accept `Path` from pipeline, current version does not support this. This dost cause breaking change.
To pass the CI please add this to Breakingchange.csv.

**Comment by wyunchi-ms** [comment]
And please resolve the conflicts. @blueww

**Pull request merged**